### PR TITLE
CO-72 Order refill.

### PR DESCRIPTION
**Pull request by JeppeTrip**

### CODE REVIEW

File: backend/Web/Controllers/RefillController.cs [hunk 1]
```diff
@@ -36,13 +37,19 @@ public async Task<ActionResult<int>> Create(CreateRefillCommand command)
...
-            RefillId = id
-        });
+      return await Mediator.Send(new SaveCouponImageCommand
+      {
+        File = file,
+        RefillId = id
+      });
+    }
+
+    [HttpPost("/OrderRefill")]
```
Comment (JeppeTrip) [inline backend/Web/Controllers/RefillController.cs:49]: Do we want the command parameters in the api route?

File: backend/Application/Refills/Commands/OrderRefill/OrderRefillCommand.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if(location == null)
+        {
+          throw new ArgumentException("Nonexistent location: "+request.LocationId);
+        }
+
+        var refill = route.Refills.FirstOrDefault(refill => refill.LocationId == request.LocationId);
+        if(refill != null)
+        {
+          route.Refills.Remove(refill);
+        }
```
Comment (objmj-itminds) [inline backend/Application/Refills/Commands/OrderRefill/OrderRefillCommand.cs:50]: @CSCITM what's your thoughts here? Would you rather query on the refill DBSet directly?

File: backend/Web/Controllers/RefillController.cs (hunk 1 above)
Comment (objmj-itminds) [reply backend/Web/Controllers/RefillController.cs:49]: Nope. The command is translated into being the required body.

**Review by objmj-itminds — APPROVED** [review]
Approved as a default, minor adjustments to make and to consider for the future.

File: backend/Application/Refills/Commands/OrderRefill/OrderRefillCommand.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      public OrderRefillCommandHandler(IApplicationDbContext context)
+      {
+        _context = context;
+      }
+
+      public async Task<int> Handle(OrderRefillCommand request, CancellationToken cancellationToken)
+      {
+        var route = _context.Routes
+          .Include(x => x.Refills)
+          .FirstOrDefault(route => route.Id == request.RouteId);
```
[the remark this answers is not in the document]
Comment (JeppeTrip) [reply backend/Application/Refills/Commands/OrderRefill/OrderRefillCommand.cs:34]: If I'm doing the include FindAsync isn't an option, it just gives me this error:
`'IIncludableQueryable<Route, ICollection<Refill>>' does not contain a definition for 'FindAsync' and no accessible extension method 'FindAsync' accepting a first argument of type 'IIncludableQueryable<Route, ICollection<Refill>>' could be found (are you missing a using directive or an assembly reference?) [Application]csharp(CS1061)`

**Pull request merged**